hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -441,15 +441,15 @@ def _get_line_name(self, project, task=None):
             name += ' - {}'.format(task.name)
         return name
 
-    def _get_default_analytic_line(self, date, project, task, timesheet=None):
+    def _get_default_sheet_line(self, date, project, task, timesheets=None):
         name_y = self._get_line_name(project, task)
-        timesheet = self.clean_timesheets(timesheet)
+        timesheet = self.clean_timesheets(timesheets)
         values = {
             'value_x': self._get_date_name(date),
             'value_y': name_y,
             'date': date,
             'project_id': project.id,
-            'task_id': task and task.id or False,
+            'task_id': task and task.id,
```

Comment:
won't task.id resolve to False if task is empty?

Reply:
yes. What's the problem?

Reply:
I mean, won't `'task_id': task.id,` be enough?

Reply:
if `task` is not an empty record but is a `False` or `None`, then it breaks. It's common to do this `and` to avoid possible breaks just in case.